alvr/xtask/src/format.rs

```diff
@@ -1,52 +1,40 @@
 use alvr_filesystem as afs;
 use std::fs;
 use std::mem;
-use walkdir::{DirEntry, FilterEntry, WalkDir};
+use std::path::PathBuf;
+use walkdir::WalkDir;
 use xshell::{cmd, Shell};
 
-fn should_format(entry: &DirEntry) -> bool {
-    if entry.file_type().is_dir() {
-        match entry.path().file_name().unwrap().to_str().unwrap() {
-            "shared" | "include" => false,
-            _ => true,
-        }
-    } else {
-        let is_cpp_file = entry
-            .path()
-            .extension()
-            .is_some_and(|ext| matches!(ext.to_str().unwrap(), "c" | "cpp" | "h" | "hpp"));
-
-        let should_skip = entry
-            .path()
-            .strip_prefix(afs::workspace_dir().join("alvr"))
-            .unwrap()
-            .to_str()
-            .is_some_and(|name| {
-                matches!(
-                    name,
-                    "server/cpp/platform/win32/NvCodecUtils.h"
-                        | "server/cpp/platform/win32/NvEncoder.cpp"
-                        | "server/cpp/platform/win32/NvEncoder.h"
-                        | "server/cpp/platform/win32/NvEncoderD3D11.cpp"
-                        | "server/cpp/platform/win32/NvEncoderD3D11.h"
-                        | "server/cpp/alvr_server/nvEncodeAPI.h"
-                )
-            });
-
-        is_cpp_file && !should_skip
-    }
-}
-
-fn get_files() -> std::iter::Filter<
-    FilterEntry<walkdir::IntoIter, impl FnMut(&DirEntry) -> bool>,
-    impl FnMut(&Result<DirEntry, walkdir::Error>) -> bool,
-> {
-    let cpp_dir = afs::crate_dir("server").join("cpp/");
+fn files_to_format_paths() -> Vec<PathBuf> {
+    let cpp_dir = afs::crate_dir("server").join("cpp");
 
     WalkDir::new(cpp_dir)
         .into_iter()
-        .filter_entry(|e| should_format(e))
-        .filter(|e| !e.as_ref().unwrap().file_type().is_dir())
+        .filter_entry(|entry| {
```

Comment:
Also broken, since `filter_entry` also gets the intermediate folders in a path, i.e. `alvr_server` for `alvr_server/alvr_server.cpp`, which gets excluded because it's not included which means none of the files in `alvr_server` get considered. You have to do the logic like I had it